plugins/modules/mysql_db.py

```diff
@@ -422,13 +430,20 @@ def db_dump(module, host, user, password, db_name, target, all_databases, port,
     elif os.path.splitext(target)[-1] == '.xz':
         path = module.get_bin_path('xz', True)
 
-    if path:
+    if pipefail and path:
+        cmd = 'set -o pipefail && %s | %s > %s' % (cmd, path, shlex_quote(target))
+    elif path:
         cmd = '%s | %s > %s' % (cmd, path, shlex_quote(target))
```

Comment:
```suggestion
    if path:
        cmd = '%s | %s > %s' % (cmd, path, shlex_quote(target))
        if pipefail:
            cmd = 'set -o pipefail && ' + cmd
```
how about? As path is true in both the cases, we could make it a bit less verbose

Reply:
I think your solution is better. I try to avoid nested if but here it make sense because it lessen repetitions.

One question thought: What is the minimum Python version supported? If it is 3.6+ then I propose to convert line 434 to fstring.

Reply:
Why this conversation is marked resolved?
Does that means that I can go ahead and convert to fstring?

Reply:
@laurent-indermuehle ah, sorry, i thought you committed it:) Now i see. I think practically it can work with older versions, so if there's no strong reason to use incompatible things, i would avoid this.

Reply:
@Andersson007 Ok, then no fstring. Great, I commited your changes. I think we are done with this PR, unless the CI tells otherwise.

I'm not 100% satisfied because, the default is highly insecure or prevent us from using compression. If a way exists to pipe commands together with Python subcommand.Popen() while preserving the return code of each command, it would be ways better. But I failed to find it. Especially in our case, where this method is wrapped by Ansible module.run_command().

Without help from an expert I can't offer a better solution.

I was happy to help, I learned alot and your guidances was very appreciated.